// @vymalo/ui-preset/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const defaultTheme_1 = require("tailwindcss/defaultTheme");
const colord_1 = require("colord");
const colors_1 = require("@mui/material/colors");
const mainColor = colors_1.purple;
const coreVymaloTheme = {
    primary: mainColor[500],
    'primary-content': mainColor[50],
    secondary: colors_1.teal[400],
    'secondary-content': colors_1.teal[100],
    accent: colors_1.green[500],
    'accent-content': colors_1.green[50],
    neutral: colors_1.grey[900],
    'neutral-content': colors_1.grey[800],
    'base-100': (0, colord_1.colord)(mainColor[900]).darken(0.2).desaturate(0.45).toHex(),
    'base-200': (0, colord_1.colord)(mainColor[900]).darken(0.215).desaturate(0.47).toHex(),
    'base-300': (0, colord_1.colord)(mainColor[900]).darken(0.23).desaturate(0.5).toHex(),
    'base-content': (0, colord_1.colord)(mainColor[50]).lighten(0.05).toHex(),
    info: colors_1.blue[700],
    'info-content': colors_1.blue[50],
    success: colors_1.green[600],
    'success-content': colors_1.green[50],
    warning: colors_1.orange[800],
    'warning-content': colors_1.orange[100],
    error: colors_1.red[800],
    'error-content': colors_1.red[50],
};
const lightVymaloTheme = {
    primary: mainColor[400],
    'primary-content': mainColor[50],
    secondary: colors_1.teal[700],
    'secondary-content': colors_1.teal[100],
    accent: colors_1.green[400],
    'accent-content': colors_1.green[50],
    neutral: colors_1.grey[50],
    'neutral-content': colors_1.grey[900],
    'base-100': (0, colord_1.colord)(mainColor[50]).lighten(0.04).desaturate(0.3).toHex(),
    'base-200': (0, colord_1.colord)(mainColor[50]).lighten(0.02).desaturate(0.4).toHex(),
    'base-300': (0, colord_1.colord)(mainColor[50]).lighten(0.01).desaturate(0.5).toHex(),
    'base-content': mainColor[900],
    info: colors_1.blue[400],
    'info-content': colors_1.blue[50],
    success: colors_1.green[500],
    'success-content': colors_1.green[50],
    warning: colors_1.orange[600],
    'warning-content': colors_1.orange[50],
    error: colors_1.red[500],
    'error-content': colors_1.red[50],
};
const basePreset = {
    theme: {
        extend: {
            fontSize: {
                '3xl': '2rem',
            },
            fontFamily: {
                sans: ['Louis George Cafe', ...defaultTheme_1.fontFamily.sans],
                serif: ['Louis George Cafe', ...defaultTheme_1.fontFamily.serif],
                mono: ['Louis George Cafe', ...defaultTheme_1.fontFamily.mono],
            },
        },
    },
    plugins: [
        require('@tailwindcss/aspect-ratio'),
        require('tailwind-extended-shadows'),
        require('@tailwindcss/typography'),
        require('tailwindcss-radix')({}),
        require('@tailwindcss/forms')({ strategy: 'class' }),
        require('daisyui'),
    ],
    daisyui: {
        themes: [
            {
                'vymalo-light': lightVymaloTheme,
            },
            {
                'vymalo-dark': coreVymaloTheme,
            },
        ],
    },
    darkMode: ['class', '[data-theme="vymalo-dark"]'],
};
exports.default = basePreset;
